llvm/lib/Support/Signals.cpp

```diff
@@ -253,6 +253,122 @@ static bool printSymbolizedStackTrace(StringRef Argv0, void **StackTrace,
   return true;
 }
 
+#if ENABLE_DEBUGLOC_ORIGIN_TRACKING
+void sys::symbolizeAddresses(AddressSet &Addresses,
+                             SymbolizedAddressMap &SymbolizedAddresses) {
+  assert(!DisableSymbolicationFlag && !getenv(DisableSymbolizationEnv) &&
+         "Debugify origin stacktraces require symbolization to be enabled.");
+
+  // Convert Set of Addresses to ordered list.
+  SmallVector<void *, 0> AddressList(Addresses.begin(), Addresses.end());
+  if (AddressList.empty())
+    return;
+  int NumAddresses = AddressList.size();
+  llvm::sort(AddressList);
+
+  // Use llvm-symbolizer tool to symbolize the stack traces. First look for it
```

Comment:
It seems like this copies code from `printSymbolizedStackTrace` above...? Can we rework this to share common code instead of duplicating it?

Reply:
Possibly - my goal to start off here was to keep this as "out of the way" as possible, by purely inserting new code into conditionally compiled blocks whenever possible; I can see if there's a clean way to share the code.

Reply:
I don't see a need to be so conservative with this particular bit in the `Support` library. `symbolizeAddresses` seems like a useful function that others may want to use as well. So, I'd encourage making that available all the time (removing the `#if` guards) and refactoring to deduplicate shared code here.